command/plugins_required.go

```diff
@@ -0,0 +1,109 @@
+package command
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"runtime"
+	"strings"
+
+	pluginsdk "github.com/hashicorp/packer-plugin-sdk/plugin"
+	plugingetter "github.com/hashicorp/packer/packer/plugin-getter"
+	"github.com/mitchellh/cli"
+)
+
+type PluginsRequiredCommand struct {
+	Meta
+}
+
+func (c *PluginsRequiredCommand) Synopsis() string {
+	return "List plugins required by a config"
+}
+
+func (c *PluginsRequiredCommand) Help() string {
+	helpText := `
+Usage: packer plugins required <path>
+
+  This command will list every Packer plugin required by a Packer config.
+  All binaries matching the required version constrain and the current OS and 
+  Architecture will be listed.
+
+
+  Ex: packer plugins required require.pkr.hcl
+  Ex: packer plugins required path/to/folder/
```

Comment:
When running against a JSON template I get an error, which is what I expected. But I think it might be good to update the error to mention the `packer plugins install` command. What do you think? 

```
~>  packer plugins required ../packer-templates/amazon-ebs/amazon-ebs_secretsmanager_shell-local.json
Error: Packer init is supported for HCL2 configuration templates only

Please manually install plugins or use a HCL2 configuration that will do that
for you.
```

Reply:
Good catch !

Reply:
I updated this.